core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -161,42 +184,22 @@
       // invalid scheme. Just fall through and will try "dns:///foo.googleapis.com:8080"
       uriSyntaxErrors.append(e.getMessage());
     }
-    if (nameResolver == null) {
+    // It may be Opaque if the host was parsed as the scheme
+    if (targetUri == null || targetUri.isOpaque()) {
```

Comment:
If the name resolver is going to have to dig into the URI anyways, why not let it handle the URI stuff and not do it here?  Wouldn't it be better to not pass around a bad URI (one representing host://...) ?  Like, if the name resolver is the one deciding if a target string is usuable, why go through the pains of parsing it here?